usr/src/lib/brand/lx/lx_init/lxinit.c

```diff
@@ -700,31 +713,55 @@ lxi_net_setup(zone_dochandle_t handle)
 }
 
 static void
-lxi_net_static_route(const char *line)
+lxi_net_setup_gateways(zone_dochandle_t handle)
 {
-	/*
-	 * Each static route line is a string of the form:
-	 *
-	 *	"10.77.77.2|10.1.1.0/24|false"
-	 *
-	 * i.e. gateway address, destination network, and whether this is
-	 * a "link local" route or a next hop route.
-	 */
-	custr_t *cu = NULL;
-	char *gw = NULL, *dst = NULL;
-	int pfx = -1;
-	int i;
+	struct zone_nwiftab lookup;
 
-	if (custr_alloc(&cu) != 0) {
-		lxi_err("custr_alloc failure");
+	if (zonecfg_setnwifent(handle) != Z_OK)
+		return;
+
+	while (zonecfg_getnwifent(handle, &lookup) == Z_OK) {
+		const char *iface = lookup.zone_nwif_physical;
+		struct zone_res_attrtab *attrs = lookup.zone_nwif_attrp;
+		const char *primary, *gateway;
+
+		if (zone_find_attr(attrs, "primary", &primary) == 0 &&
+		    strcmp(primary, "true") == 0 &&
+		    zone_find_attr(attrs, "gateway", &gateway) == 0) {
+			lxi_iface_gateway(iface, NULL, 0, gateway, B_FALSE);
+		}
 	}
 
+	(void) zonecfg_endnwifent(handle);
+}
+
+/*
+ * Parse a static route line string of the form:
+ *
+ *      "10.77.77.2|10.1.1.0/24|false"
+ *
+ * i.e. gateway address, destination network, and whether this is
+ * a "link local" route or a next hop route.
+ *
+ * passing a true value  to the llonly argument to this method causes only
```

Comment:
Does this comment need updating? I don't see a `llonly` argument for the method signature below.

Reply:
It might need to be moved down to lxi_net_process_route_line().

Reply:
This comment was actually on lxi_net_process_route_line(), but during a previous review we felt it might make more sense where it ended up. It seems I only partially reworded things at the time. I can either move it or fix the wording if anyone has a strong opinion one way or the other.

Reply:
Doing something is better than doing nothing.  Pick and I'm pretty sure we'll be okay with it.

Reply:
I've removed the sentence in question; even in its previous position it was sort of re-stating the obvious.